repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasGraphUtilsV2.java

```diff
@@ -478,6 +478,93 @@ public static boolean typeHasInstanceVertex(AtlasGraph graph, String typeName) t
+     * either directly or through propagation.
+     * 
+     * @param typeName the classification type name to check
+     * @return true if any active entities have this classification, false otherwise
+     * @throws AtlasBaseException if the ES query fails
+     */
+    public static boolean classificationHasReferences(String typeName) throws AtlasBaseException {
+        return classificationHasReferences(getGraphInstance(), typeName);
+    }
+
+    /**
+     * Check if a classification type has references using Elasticsearch.
+     * This method searches for entities that have the classification attached
+     * either directly or through propagation.
+     * 
+     * @param graph the AtlasGraph instance
+     * @param typeName the classification type name to check
+     * @return true if any active entities have this classification, false otherwise
+     * @throws AtlasBaseException if the ES query fails
+     */
+    public static boolean classificationHasReferences(AtlasGraph graph, String typeName) throws AtlasBaseException {
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Checking if classification {} has references using ES", typeName);
+            }
+
+            String indexName = Constants.getESIndex();
+            AtlasIndexQuery indexQuery = graph.elasticsearchQuery(indexName);
+
+            String esQuery = buildClassificationReferenceQuery(typeName);
+            Long count = indexQuery.countIndexQuery(esQuery);
+            
+            boolean hasReferences = count != null && count > 0;
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Classification {} has references: {} (count: {})", typeName, hasReferences, count);
+            }
+
+            return hasReferences;
+        } catch (Exception e) {
+            LOG.error("Error checking classification references for {}: {}", typeName, e.getMessage(), e);
+            // Conservative approach: if ES query fails, assume there are references to prevent accidental deletion
+            return true;
+        }
+    }
+
+    /**
+     * Build Elasticsearch query to check for classification references.
+     * Searches for entities that have the classification in either __traitNames or __propagatedTraitNames.
+     * 
+     * @param typeName the classification type name
+     * @return JSON query string
+     */
+    private static String buildClassificationReferenceQuery(String typeName) {
+        // Escape the type name for JSON
+        String escapedTypeName = typeName.replace("\"", "\\\"").replace("\\", "\\\\");
+        
+        return String.format(
+            "{\n" +
+            "  \"query\": {\n" +
+            "    \"bool\": {\n" +
+            "      \"filter\": [\n" +
+            "        {\n" +
+            "          \"bool\": {\n" +
+            "            \"should\": [\n" +
+            "              {\"terms\": {\"%s\": [\"%s\"]}},\n" +
```

Comment:
No need of `terms`, pass replace it with `term` as we only want to filter a single tag at a time